### PR TITLE
update SX

### LINKED ISSUE #33: Fix case when a location has not invariant (or flow)
If a location has not defined an invariant field, as in the example below, the parser currently crashes, since `I` (invariant) is not defined in the [parse_location](https://github.com/JuliaReach/SX.jl/blob/master/src/parse.jl#L297) function. 

```julia
    <location id="1" name="running" x="251.0" y="209.0" width="146.0" height="140.0">
      <flow>x'== 10 + (y - x) &amp;
y'== x * (28 - z) - y &amp; 
z'== x * y - 2.6667*z</flow>
    </location>
``` 

Instead, if there is no location it should return an empty vector of expressions.

### PR DESCRIPTION
**Pull request by mforets**
Closes #33.

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,4 +1,4 @@
-julia 0.6
+julia 1.1
```
Comment (schillic) [inline REQUIRE]: I would not require versions > v1.0. Many people will use that version. You even "highly recommended" that recently and the text in `README` says v1.0 :wink:
(For the same reason, do not drop the version in Travis.)

File: README.md [hunk 2]
```diff
@@ -12,25 +12,30 @@
...
 
 ## Updating
 
-To checkout the latest version, do
+To checkout the latest (development) version, do:
+
 ```julia
-Pkg.checkout("SX")
-````
+Pkg.add(PackageSpec(url="https://github.com/JuliaReach/SX.jl", rev="master"))
```
Comment (schillic) [inline README.md]: Is `rev="master"` not the default?

File: src/parse.jl [hunk 3]
```diff
@@ -287,24 +289,38 @@ end
...
         elseif nodename(element) == "flow"
-            f = parse_sxmath(nodecontent(element))
+            flow = parse_sxmath(nodecontent(element))
         else
             @warn("field $(nodename(element)) in location $(field["id"]) is ignored")
         end
     end
-    return (id, I, f)
+
+    # if no invariant is defined in the model => return empty expressions
```
Comment (schillic) [inline src/parse.jl]:
```suggestion
    # if no invariant is defined in the model => return empty list of expressions
```

File: src/parse.jl [hunk 4]
```diff
@@ -287,24 +289,38 @@ end
...
+    if !@isdefined(invariant) || invariant isa Vector{Nothing}
+        invariant = Expr[:()]
+    end
+
+    # if no flow is defined in the component => return empty expressions
```
Comment (schillic) [inline src/parse.jl]:
```suggestion
    # if no flow is defined in the component => return empty list expressions
```

File: src/parse.jl [hunk 5]
```diff
@@ -287,24 +289,38 @@ end
...
+            flow = parse_sxmath(nodecontent(element))
         else
             @warn("field $(nodename(element)) in location $(field["id"]) is ignored")
         end
     end
-    return (id, I, f)
+
+    # if no invariant is defined in the model => return empty expressions
+    if !@isdefined(invariant) || invariant isa Vector{Nothing}
+        invariant = Expr[:()]
```
Comment (schillic) [inline src/parse.jl]: What does `:()` do? :frog:

File: README.md (hunk 2 above)
Comment (mforets) [reply README.md]: Hmm i don't know, probably yes? This use is one of the examples in the docstring,

```julia
  Examples
  ≡≡≡≡≡≡≡≡≡≡

  Pkg.add("Example") # Add a package from registry
  Pkg.add(PackageSpec(name="Example", version="0.3")) # Specify version
  Pkg.add(PackageSpec(url="https://github.com/JuliaLang/Example.jl", rev="master")) # From url
  Pkg.add(PackageSpec(url="/remote/mycompany/juliapackages/OurPackage"))` # From path (has to be a gitrepo)
```

File: src/parse.jl (hunk 5 above)
Comment (mforets) [reply src/parse.jl]: It's an empty expression

```julia
julia> :()
:(())

julia> typeof(ans)
Expr
```

Comment (schillic) [reply src/parse.jl]: So why not define `invariant = Expr[]`?

File: REQUIRE (hunk 1 above)
Comment (schillic) [reply REQUIRE]: By the way, I think the `REQUIRE` file is redundant if there is a `Project` file...

Comment (mforets) [reply REQUIRE]: Yes, that's true. Should i delete the REQUIRE then?

Comment (schillic) [reply REQUIRE]: If it works, yes!

Comment (schillic) [reply REQUIRE]: But wait! The `Project` file does not restrict the Julia version.

Comment (mforets) [reply REQUIRE]: Yes, i was thinking about that. Seems like [it has to be specified under the [compat] tag](https://julialang.github.io/Pkg.jl/v1/compatibility/#**6.**-Compatibility-1) as in: 

```julia
[compat]
julia = "1.0"
```

**Pull request merged**